### PR TITLE
[chore] Add actionlint to CI, and fix issues it detects

### PR DESCRIPTION
**Pull request by danielsn**
# What does this PR do?

Uses the actionlint CI action, and fixes issues it found.

# Motivation
Ironically, `actionlint` found an issue in this linting action!
```
╰─➤  actionlint                                                                         
../.github/workflows/lint.yml:33:27: property "version" is not defined in object type {platform: string; rust_version: string} [expression]
   |
33 |       - name: Install ${{ matrix.version }} toolchain and clippy
```
# Additional Notes

None of these issues seemed serious, but worth fixing and checking in CI. 
There were a bunch of violations of proper quoting, enough that I'll leave them for a future PR
# How to test the change?

Describe here in detail how the change can be validated.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -30,11 +40,11 @@ jobs:
         uses: ./.github/actions/cache
         with:
           rust_version: ${{ matrix.rust_version }}
-      - name: Install ${{ matrix.version }} toolchain and clippy
+      - name: Install ${{ matrix.rust_version }} toolchain and clippy
         run: rustup install ${{ matrix.rust_version }} && rustup default ${{ matrix.rust_version }} && rustup component add clippy
       - name: Run clippy on ${{ matrix.platform }} ${{ matrix.rust_version }} 
         shell: bash
-        run: cargo clippy --all-targets --all-features -- -D warnings $([ ${{ matrix.rust_version }} = 1.71.1 ] && echo -Aunknown-lints)
+        run: cargo clippy --all-targets --all-features -- -D warnings "$([ ${{ matrix.rust_version }} = 1.71.1 ] && echo -Aunknown-lints)"
```
Comment (danielsn) [inline .github/workflows/lint.yml:47]: Can someone with expert level bash-fu confirm this does what we want?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -45,7 +55,7 @@ jobs:
         run: sudo apt-get install -y licensecheck
       - name: Check licenses
         # Exclude symbolizer-ffi from the checks (mostly imported code)
-        run: '! find . \( -name "*.rs" -o -name "*.c" -o -name "*.sh" \)  -not -path "./symbolizer-ffi/*" -not -path "*/tarpc/*" | xargs licensecheck -c ".*" | grep -v "Apache License 2.0"'
+        run: '! find . \( -name "*.rs" -o -name "*.c" -o -name "*.sh" \)  -not -path "./symbolizer-ffi/*" -not -path "*/tarpc/*" -print0 | xargs -0 licensecheck -c ".*" | grep -v "Apache License 2.0"'
```
Comment (danielsn) [inline .github/workflows/lint.yml:58]: https://www.shellcheck.net/wiki/SC2038

File: .github/workflows/lint.yml (hunk 1 above)
Comment (danielsn) [reply .github/workflows/lint.yml:47]: https://www.shellcheck.net/wiki/SC2046

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
 jobs:
+  actionlint: 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+      - name: Run actionlint
+        uses: devops-actions/actionlint@v0.1.3
+        with:
+          shellcheck_opts: '-e SC2086'
```
Comment (danielsn) [inline .github/workflows/lint.yml:16]: A bunch of violations of this https://www.shellcheck.net/wiki/SC2086 which I'm going to postpone to a future PR

**Review by ekump — APPROVED** [review]
My bash-fu is not expert level, but LGTM

**Pull request merged**